sharktank/sharktank/layers/configs/llm_configs.py

```diff
@@ -41,37 +42,51 @@ class LlamaHParams:
     attention_head_count_kv: int
     expert_count: int
     expert_used_count: int
+    model_variant: str
 
     @staticmethod
     def from_gguf_props(p: dict[str, Any]):
+        name_prefix = "llama"
+        # These shouldnt be required for llama.
+        default_attn_dim = None
+        default_rope_count = None
+        if "grok.attention.head_count" in p:
```

Comment:
Let's branch on the gguf value `general.architecture == "grok"` so that we can support more single headed models than just llama and more multiheaded models than just grok.

Reply:
I have this already in my stream of commits. Will push in the next 30 mins.